service/history/mutableStateBuilder.go

```diff
@@ -621,14 +732,30 @@ func (e *mutableStateBuilder) assignTaskIDToEvents() error {
 }
 
 func (e *mutableStateBuilder) IsCurrentWorkflowGuaranteed() bool {
```

Comment:
Not clear what "Guaranteed" means in this context. I can sort of guess based on the code. Might worth putting comments on this.